### PR TITLE
Wrote documentation for `NodeEncoder` class

### PR DESCRIPTION
**Pull request by nh916**
# Description
Wrote documentation for `NodeEncoder` class

## Screenshots
image

## Checklist

- [ ] My name is on the list of contributors (`CONTRIBUTORS.md`) in the pull request source branch.
- [x] I have updated the documentation to reflect my changes.

### CODE REVIEW

File: src/cript/nodes/util/__init__.py [hunk 1]
```diff
@@ -22,12 +22,80 @@
...
+    condensed representations to avoid redundancy in the JSON output.
+    It also allows suppressing specific attributes from being included in the serialized JSON.
+
+    Attributes
+    ----------
+    handled_ids : Set[str]
+        A set to store the UIDs of nodes that have been processed during serialization.
+    known_uuid : Set[str]
+        A set to store the UUIDs of nodes that have been previously encountered in the JSON.
+    condense_to_uuid : Set[str]
```
Comment (InnocentBug) [inline src/cript/nodes/util/__init__.py:38]: Ok, I think this is my fault.

This isn't actually the type of `condense_to_uuid` the actual type is `Dict[str, List[str]]`, with the key being the name of the node, and the value a list (maybe we should make it a Set) of the attributes.

Comment (nh916) [reply src/cript/nodes/util/__init__.py:38]: whatever you think is best works with me

Comment (InnocentBug) [reply src/cript/nodes/util/__init__.py]: i fixed it

Comment (nh916) [reply src/cript/nodes/util/__init__.py]: Thank you so much!

**Pull request merged**